src/test/fakeExtensionContext.ts

```diff
@@ -41,7 +42,10 @@ export class FakeExtensionContext implements vscode.ExtensionContext {
     public storagePath: string | undefined
     public logPath: string = ''
 
-    private _extensionPath: string = ''
+    // Seems to be the most reliable way to set the extension path (unfortunately)
+    // TODO: figure out a robust way to source the project directory that is invariant to entry point
+    // Using `package.json` as a reference point seems to make the most sense
+    private _extensionPath: string = resolve(__dirname, '../../..')
```

Comment:
Best place might be the Mocha entry point since it's supposed to be for tests only. In a real environment you'll have the extension context.

(I need to address all the other feedback...)